### PR TITLE
Add an example clojure project

### PR DESCRIPTION
**Pull request by whentze**
This builds already, but I want to add some tests based on this as well.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -57,6 +57,8 @@ in {
           # Ensures that clojure creates all the caches in our empty separate home directory
           export JAVA_TOOL_OPTIONS="-Duser.home=$tmp/home"
 
+          mkdir -p "$tmp/home/.gitlibs/libs"
```
Comment (infinisil) [inline default.nix]: Why is this necessary?

Comment (whentze) [reply default.nix]: `locker.py` expects it to be there, and if there are 0 git dependencies then the clojure tooling never creates it.

Thinking about it though, maybe we want `locker.py` to handle this gracefully instead and maybe not even emit a `git` key in the lockfile then. WDYT?

Comment (infinisil) [reply default.nix]: Yeah I think handling this in the python script would be better

Comment (whentze) [reply default.nix]: I'll do that instead and push it :)

**Review by infinisil — APPROVED** [review]
Looking good!
